File: frontend/src/services/api/publicInvitations.ts
import { fetchApi } from "../../lib/api";

export interface InvitationPublicContext {
  job_title: string;
  job_description?: string;
  seniority?: string;
  candidate_instructions?: string;
  duration_minutes: number;
  invitation_status: string;
  candidate_email: string;
}

export interface RedeemedSessionInfo {
  id: string;
  job_id?: string;
  definition_id?: string;
  status: string;
  created_at?: string;
}

export interface RedeemResponse {
  session: RedeemedSessionInfo;
  livekit_token: string;
  livekit_url: string;
}

export async function getInvitationContext(token: string): Promise<InvitationPublicContext> {
  return fetchApi<InvitationPublicContext>(`/api/v1/invitations/${token}`);
}

export async function redeemInvitation(token: string): Promise<RedeemResponse> {
  return fetchApi<RedeemResponse>(`/api/v1/invitations/${token}/redeem`, {
    method: "POST",
  });
}

File: backend/backend/schemas/public_invitations.py
"""Public (no-auth / candidate-JWT) schemas for Phase 6, Sub-phase 6B."""
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime
from uuid import UUID


class InvitationPublicContext(BaseModel):
    """What a candidate sees on the invite landing page — deliberately not
    the full admin JobResponse shape, to avoid leaking internal fields."""
    job_title: str
    job_description: Optional[str] = None
    seniority: Optional[str] = None
    candidate_instructions: Optional[str] = None
    duration_minutes: int
    invitation_status: str
    candidate_email: str


class RedeemedSessionInfo(BaseModel):
    id: UUID
    job_id: Optional[UUID] = None
    definition_id: Optional[UUID] = None
    status: str
    created_at: Optional[datetime] = None


class RedeemResponse(BaseModel):
    session: RedeemedSessionInfo
    livekit_token: str
    livekit_url: str
